dali/kernels/tensor_view.h

```diff
@@ -414,6 +414,32 @@ TensorListView<StorageGPU, T, ndim> make_tensor_list_gpu(T *data, TensorListShap
   return { data, std::move(shape) };
 }
 
+
+/**
+ * @brief Get a subtensor by slicing along outermost dimension at index `idx`
+ * 
+ * @details Produces tensor, for which number of dimensions is reduced by 1.
+ * Removed dimension is outer-most (e.g. for shape {3,2,4,6} produces {2,4,6}).
+ * Data inside the tensor is extracted according to provided index.
+ * Data is not copied.
+ *
+ * Example:
+ * tv.data = [[1, 2, 3], [4, 5, 6]]       (shape: [2, 3])
+ * oust_dimension(tv, 1) -> [4, 5, 6]     (shape: [3])
+ *
+ * @param source Source TensorView
+ * @param idx Index inside dimension, along which data is extracted
+ * @return TensorView with reduced dimensionality
+ */
+template<typename StorageBackend, typename DataType, int ndims>
+TensorView<StorageBackend, DataType, ndims - 1>
+subtensor(TensorView<StorageBackend, DataType, ndims> source, size_t idx) {
```

Comment:
use int64_t for index - that's what we use almost everywhere, this function shouldn't be different.

Reply:
Why? It's not like idx (or pos, whatever) can be <0...